components/LoadMore.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { t } from 'ttag';
+import { makeStyles } from '@material-ui/core/styles';
+import { Box, CircularProgress } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  loadMore: {
+    width: '33%',
+    color: theme.palette.secondary[300],
+    outline: 'none',
+    cursor: 'pointer',
+    borderRadius: 30,
+    padding: 10,
+    background: 'transparent',
+    border: `1px solid ${theme.palette.secondary[300]}`,
+  },
+  loading: {
+    color: theme.palette.secondary[300],
+  },
+}));
+
+export default function LoadMore({
+  load = () => {},
+  setCursor = () => {},
+  loading = false,
+  pageInfo = {},
+  edges = [],
+}) {
+  const { firstCursor, lastCursor } = pageInfo;
+  if (!firstCursor || !lastCursor) {
+    return null;
+  }
+
+  const lastCursorOfPage =
+    edges.length && edges[edges.length - 1] && edges[edges.length - 1].cursor;
+
+  const loadMore = () => {
+    if (lastCursorOfPage !== lastCursor) {
+      setCursor(lastCursorOfPage);
+      load();
```

Comment:
If we leverage `loadMore` and get cursor from last edge directly in `ArticlePageLayout`, we can leave the pagination logic entirely in where pagination happens, reduce the props passed to `LoadMore`.

Reply:
I eventually remove the LoadMore component and leverage the whole component to `AarticlePageLayout`, because with only `onClick` &  `loading` the LoadMore component seems similar to regular button, the actual logics are mostly  move to `ArticlePageLayout`, it's a bit strange to have a Component call `LoadMore` while the actual load-more logics are written in upper component.